protect-function-app/greeting/index.js

```diff
@@ -0,0 +1,20 @@
+/*
+ This Azure Function responds at GET /api/greeting.
+ 
+ Authentication and authorization are performed by Azure Functions (sometimes called "Easy Auth").
+ This function decodes the JSON web token to verify the required scope is included.
+*/
+
+// This module decodes JSON web tokens
+const jwt = require('jsonwebtoken')
+
+module.exports = async function (context, req) {
+  // Decode the JSON web token which is passed in via the Authorization header
+  const decodedToken = jwt.decode(req.headers.authorization.split(' ')[1])
+
+  // If the scope includes 'Greeting.Read', then accept the request with a greeting message and default 200 status code
+  // Otherwise reject the request with a failure message and 403 status code
+  context.res = (decodedToken.scp.includes('Greeting.Read'))
```

Comment:
`includes` doesn't seem very safe here.  If the token had `Greeting.ReadWrite` or `SystemGreeting.ReadSettings` this would return true.  I think you need to split on ` ` and then check for the specific scope, yeah?  Unless this isn't just the `string.includes()` method and something else from jwt?

Reply:
Nice catch, I will investigate.

Reply:
New version looks good.